plugins/pagerduty/src/api/client.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core';
+import {
+  Service,
+  Incident,
+  RequestOptions,
+  ClientApiConfig,
+  ServicesResponse,
+  IncidentsResponse,
+  OnCallsResponse,
+  OnCall,
+} from '../components/types';
+import { PagerDutyClient } from './types';
+
+export class UnauthorizedError extends Error {}
+
+export const pagerDutyApiRef = createApiRef<PagerDutyClient>({
+  id: 'plugin.pagerduty.api',
+  description: 'Used to fetch data from PagerDuty API',
+});
+
+export class PagerDutyClientApi implements PagerDutyClient {
+  constructor(private readonly config: ClientApiConfig) {}
```

Comment:
Not a perfect example, but this would be the general idea - https://github.com/backstage/backstage/blob/master/plugins/lighthouse/src/api.ts#L116. Something like this:

```ts
static fromConfig(
  config: Config,
  options: { discovery: PluginEndpointDiscovery; logger: Logger },
) {
  const eventsBaseUrl = config.getOptionalString('pagerDuty.eventsBaseUrl') ?? 'https://events.pagerduty.com/v2';
  return new PagerDutyClient({
    eventsBaseUrl,
    ...options,
  });
}
```

NOTE: I would also probably renamed `eventsUrl` to `eventsBaseUrl` to be consistent with other existing common cases. This also means that the `app-config.yaml` would need to support an optional `pagerDuty.eventsBaseUrl` option which would then be either read in from the `Config` object or defaulted.